tools/inventory-extract-windows/main.go

```diff
@@ -0,0 +1,175 @@
+var driveNumber = flag.Uint("drive", 0, "open drive with provided number")
+var drivesList = flag.Bool("list", false, "list drives")
+var helpFlag = flag.Bool("help", false, "help")
+
+func main() {
+	flag.Parse()
+	if *helpFlag {
+		fmt.Println("This program do export of INVENTORY partition from EVE drive")
+		fmt.Println("You can see list of available drives with -list flag")
+		fmt.Println("Then you must set Index of drive to -drive flag")
+		fmt.Println("With -dir flag you can provide directory to save files into. " +
+			"The program will create directory in temp if you will not provide this flag.")
+		os.Exit(0)
+	}
+	if *driveNumber == 0 {
+		fmt.Println("Please provide -drive flag with Index of device from the list below:")
+		*drivesList = true
+	}
+	if *drivesList {
+		res, err := exec.Command("wmic", "diskdrive",
+			"where", "MediaType='Removable Media'",
+			"get", "index,caption").CombinedOutput()
+		if err != nil {
+			log.Println(err)
+		} else {
+			fmt.Println(string(res))
+		}
+		time.Sleep(sleepTime)
+		os.Exit(0)
+	}
+	fd0, err := os.Open(`\\.\PhysicalDrive0`)
+	if err != nil {
+		log.Println("This program must be run by Administrator")
+		log.Println("We will ask for permissions")
+		time.Sleep(sleepTime)
+		runMeElevated()
+		os.Exit(0)
+	}
+	_ = fd0.Close()
+	f, err := os.Open(fmt.Sprintf(`\\.\PhysicalDrive%d`, *driveNumber))
+	if err != nil {
+		log.Println(err)
+		time.Sleep(sleepTime)
+		os.Exit(1)
+	}
+	defer f.Close()
+
+	table, err := partition.Read(f, 512, 512)
+
+	if err != nil {
+		log.Println(err)
+		time.Sleep(sleepTime)
+		os.Exit(1)
+	}
+	partitions := table.GetPartitions()
+	found := false
+	for _, el := range partitions {
+		fsCurrent, err := fat32.Read(f, el.GetSize(), el.GetStart(), 512)
+		if err != nil {
+			continue
+		}
+		if fsCurrent.Label() == "INVENTORY" {
+			found = true
+			var dir string
+			if *outputPath != "" {
+				dir, err = filepath.Abs(*outputPath)
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+				err = os.MkdirAll(dir, os.ModeDir)
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+			} else {
+				dir, err = ioutil.TempDir("", "INVENTORY")
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+			}
+			log.Printf("Will save to directory: %s", dir)
+			err = iterateAndSave(fsCurrent, dir, "/")
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+		}
+	}
+	if found {
+		log.Println("Processing done")
+	} else {
+		log.Println("No INVENTORY found")
+	}
+	time.Sleep(sleepTime)
+}
+
+func iterateAndSave(fsCurrent *fat32.FileSystem, dirToSave, dir string) error {
+	d, err := fsCurrent.ReadDir(dir)
+	if err != nil {
+		return err
+	}
+	for _, curFileInfo := range d {
+		if curFileInfo.Name() == "." || curFileInfo.Name() == ".." {
+			continue
+		}
+		log.Println(curFileInfo.Name())
+		if curFileInfo.IsDir() {
+			curDirPath := filepath.Join(dirToSave, curFileInfo.Name())
+			err = os.Mkdir(curDirPath, os.ModeDir)
```

Comment:
Since we are allowing output dir to exist beforehand, should we use `MkdirAll` instead? Seems like `Mkdir` is not going to like if the `curDirPath` already exists:
```go
   252  // Mkdir creates a new directory with the specified name and permission
   253  // bits (before umask).
   254  // If there is an error, it will be of type *PathError.
   255  func Mkdir(name string, perm FileMode) error {
   256  	if runtime.GOOS == "windows" && isWindowsNulName(name) {
   257  		return &PathError{Op: "mkdir", Path: name, Err: syscall.ENOTDIR}
   258  	}
   259  	longName := fixLongPath(name)
   260  	e := ignoringEINTR(func() error {
   261  		return syscall.Mkdir(longName, syscallMode(perm))
   262  	})
```

Reply:
If user provide full path to directory (he can), we need to follow whole path to it

Reply:
Ah, sorry, miss the line in the previous comment. Here we will use the directory we created before this step.
And also we will not create a directory (and skip it) if it is already exists (to not override the content).

Reply:
now I am confused. We are fine if the `outputPath` directory existed before (or did I miss the check for that), but we are are skipping all the nested pre existing dir. According to my understanding, if dir structure on `INVENTORY` looks like this:
```
INVENTORY
|-- dir1
|   |-- nested1
|   `-- some_config_file
`-- dir2
```

and `outputPath` already existed and looks like this:
```
outputPath
`-- dir1
```
Then `nested` and `some_config_file` are not going to be created

Reply:
User will be notified with problems of creating of new directories inside and will do right things he wants (clean it or preserve). I think, override anything or remove is not a good idea in our case.

Reply:
then we should check if `outputPath` existed before. `mkdirAll` will not fail if the path was there

Reply:
But it will not remove/override anything. It is the user input. User decide to save all things in ouputPath, we prepare all we need to save there.

Reply:
Not true. Current implementation will silently rewrite existing files, because:
```go
   314  // Create creates or truncates the named file. If the file already exists,
   315  // it is truncated. If the file does not exist, it is created with mode 0666
   316  // (before umask). If successful, methods on the returned File can
   317  // be used for I/O; the associated file descriptor has mode O_RDWR.
   318  // If there is an error, it will be of type *PathError.
   319  func Create(name string) (*File, error) {
   320  	return OpenFile(name, O_RDWR|O_CREATE|O_TRUNC, 0666)
   321  }
```